### DISCUSSION TITLE
How to add your language into Laravel

### DISCUSSION THREAD

**Discussion by Xoshbin** [laravel/framework#38758] [Q&A]
Hi, I have searched all around the web and Laravel's documentation. but I didn't find any articles or a repository to add a language that's not supported by Laravel framework yet, like Kurdish. except for this repository that I'm not sure if it's officially supported by the Laravel framework or not. 
So I'm asking if there's anyway to add a language that's not supported yet or if I add my language to the repository I mentioned above is it going to be added to the Laravel?
Thank you

**Comment by dennisprudlo** [comment]
What exactly do you mean with "a language that's not supported"? You can localize your app in any language.

**Reply by Xoshbin** [reply]
I mean when I change Laravel locale in config/app.php to let's say Arabic, it's changing the framework to Arabic. But Kurdish my own language is not supported yet, I want to add it. Thanks

**Reply by dennisprudlo** [reply]
What is changing? When you use for instance `$date->diffForHumans()` and it outputs "13 seconds ago" that's not laravel but a third party package. There is no file where Laravel itself gets localized.

Do you have an example in which part of Laravel the localization works with Arabic but not with Kurdish?

**Reply by Xoshbin** [reply]
Just change the locale in config/app.php to 'ar" and you see the change

**Reply by dennisprudlo** [reply]
That's not how this works. If I have a blade file with the following content:
```html
<h1>Hello</h1>
```
Changing the default locale will not magically translate the "Hello" into Arabic. You are possibly using a part of the framework or php that supports localization such as the dates or number formatting or something.

If all your content suddenly is in Arabic than you have translated it already into Arabic and your files are located in `resources/lang/ar`

**Comment by faissaloux** [comment]
Add your language inside resources/lang

**Reply by Xoshbin** [reply]
OK! So it means that Laravel only supports English by default? not any other languages?

**Reply by faissaloux** [reply]
Yes exactly. If you need other languages you have to add'em in resources/lang folder.

**Reply by Xoshbin** [reply]
Thank you @faissaloux and @dennisprudlo I knew about the way how the locale work in resource/lang, but I thought you can add other languages in the framework to have it at install.

**Reply by d-damien** [reply]
There is at least one package for it : `laravel-lang/lang`.
